### PR TITLE
Add support for backing up Collations

### PR DESCRIPTION
**Pull request by chrishajas**
Authored-by: Chris Hajas <chajas@pivotal.io>

### CODE REVIEW

File: integration/predata_shared_queries_test.go [hunk 1]
```diff
@@ -822,6 +837,26 @@ LANGUAGE SQL`)
 				resultMetadata := resultMetadataMap[oid]
 				structmatcher.ExpectStructsToMatch(&configurationMetadata, &resultMetadata)
 			})
+			It("returns a slice of default metadata for a collation in a specific schema", func() {
+				testutils.SkipIfBefore6(connection)
+				testhelper.AssertQueryRuns(connection, `CREATE COLLATION some_coll (lc_collate = 'POSIX', lc_ctype = 'POSIX');`)
+				defer testhelper.AssertQueryRuns(connection, "DROP COLLATION some_coll")
+				testhelper.AssertQueryRuns(connection, "CREATE SCHEMA testschema")
```
Comment (soumyadeep2007) [inline integration/predata_shared_queries_test.go]: Quick Q: Why do these "AssertQueryRuns" have to be defers? Can these not be asserted right after the call to the function under test?

Comment (chrishajas) [reply integration/predata_shared_queries_test.go]: The `defer` keyword ensures that the cleanup queries execute whether the test passes or fails. That way, if one test fails, other tests can still run without pollution. Go does not have a `finally`-esqe construct.

File: backup/predata_shared.go [hunk 2]
```diff
@@ -488,3 +488,13 @@ func PrintCreateDependentTypeAndFunctionAndTablesStatements(metadataFile *utils.
 		}
 	}
 }
+
+func PrintCreateCollationStatements(metadataFile *utils.FileWithByteCount, toc *utils.TOC, collations []Collation, collationMetadata MetadataMap) {
```
Comment (khuddlefish) [inline backup/predata_shared.go]: We should move this to predata_types.go

File: integration/predata_shared_queries_test.go [hunk 3]
```diff
@@ -563,6 +563,21 @@ LANGUAGE SQL`)
 				resultMetadata := resultMetadataMap[oid]
 				structmatcher.ExpectStructsToMatchExcluding(&expectedMetadata, &resultMetadata, "Oid")
 			})
+			It("returns a slice of default metadata for a collation", func() {
```
Comment (khuddlefish) [inline integration/predata_shared_queries_test.go]: Move these to predata_types_queries_test.go

File: integration/predata_shared_queries_test.go (hunk 1 above)
Comment (soumyadeep2007) [reply integration/predata_shared_queries_test.go]: Ohh! My bad, got confused thinking that the assert was related to what we were testing in the function.

File: backup/predata_shared.go (hunk 2 above)
Comment (chrishajas) [reply backup/predata_shared.go]: Whoops, meant to do that.

**Pull request merged**